### ISSUE TITLE
iOS Live Activities

### ISSUE THREAD

**Issue by NachoSoto** [jchambers/pushy#975]
Apple just released Live Activities: https://developer.apple.com/documentation/activitykit/displaying-live-data-on-the-lock-screen-with-live-activities

This will allow servers to send push notifications directly to a live activity (as opposed to the app itself): https://developer.apple.com/documentation/activitykit/activity/pushtoken

I'm still catching up on the documentation, not sure if Pushy will need to support this in any way. It's possible that just using those tokens Pushy will work for this out of the box though.

**Comment by jchambers** [comment]
I spotted this in the docs:

> Send push notifications to the Live Activity using the push token you stored on your server. It’s essential that you set the fields for the `content-state` key to match your custom [`Activity.ContentState`](https://developer.apple.com/documentation/activitykit/activity/contentstate-swift.typealias) type to ensure the system can decode the JSON payload and update the Live Activity.

I can't find any other documentation explaining where that key should be set, though. They _do_ offer an example that makes it look like it should be in the `aps` section:

```json
{
    "aps": {
        "timestamp": 1650998941,
        "event": "update",
        "content-state": {
            "driverName": "Anne Johnson",
            "estimatedDeliveryTime": 1659416400
        }
    }
}
```

…but I can't find any documented requirements about how it needs to be "shaped." My guess is that the [payload docs](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/generating_a_remote_notification) will get updated shortly. I'll plan to act when there's more concrete information available.

**Comment by rlepinski** [comment]
@jchambers I was able to get this to work today with the above payload. The topic is the main apps bundle ID + `.push-type.liveactivity` and the push-type is `liveactivity`

**Comment by jchambers** [comment]
Thanks! At this point, the concern isn't getting it to work so much as being confident that we haven't accidentally baked in some gotchas that we'd need to roll back later. I still think the right move is to wait for the documentation to catch up (it looks like it still hasn't), but I do hope to act on this soon.

**Comment by rlepinski** [comment]
@fxm90 In the console logs I found a statement from APSCourier that listed my apps bundle it with the postfix

**Comment by jchambers** [comment]
Well, the docs still haven't been updated, and it's been more than a month. I think it might be time to just go for it soon.

**Comment by rlepinski** [comment]
Just noticed they added some docs - https://developer.apple.com/documentation/activitykit/update-and-end-your-live-activity-with-remote-push-notifications. Doesn't look like the APNS payload docs have been updated yet but at least they have documented the topic and push type.

**Comment by ifrins** [comment]
I started working on Live Activities and needed support for sending push notifications, so I have created a draft PR (#990) with a base implementation for supporting Live Activities based on the few docs that are available, that seems to be working as of iOS 16.1 b2/b3
